src/pages/Fox/components/FoxTokenBalances.tsx

```diff
@@ -0,0 +1,177 @@
+import { foxAssetId, fromAssetId } from '@shapeshiftoss/caip'
+import { useMemo, useState } from 'react'
+import { AssetIcon } from 'components/AssetIcon'
+import { Text } from 'components/Text'
+import { chainIdToChainDisplayName } from 'lib/utils'
+import { AccountEntryRow } from 'pages/Accounts/components/AccountEntryRow'
+import { getFeeAssetByAssetId } from 'state/slices/assetsSlice/utils'
+import { selectRelatedAssetIds } from 'state/slices/related-assets-selectors'
+import { selectAccountIdsByChainId, selectAssets } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { useFoxPageContext } from '../hooks/useFoxPageContext'
+
+type Filter = {
+  label: string
+  chainId?: string
+  assetId?: AssetId
+}
+
+const buttonsHover = {
+  opacity: '.6',
+}
+const hstackProps: StackProps = {
+  flexWrap: {
+    base: 'wrap',
+    md: 'nowrap',
+  },
+}
+
+const ALL_FILTER_KEY = 'All'
+
+export const FoxTokenBalances = () => {
+  const buttonsBgColor = useColorModeValue('grey.500', 'white')
+  const assets = useAppSelector(selectAssets)
+  const { assetId, selectedAssetAccountId, assetAccountNumber } = useFoxPageContext()
+  const [selectedFilters, setSelectedFilters] = useState([ALL_FILTER_KEY])
+
+  const relatedAssetIdsFilter = useMemo(() => ({ assetId, onlyConnectedChains: true }), [assetId])
+  const relatedAssetIds = useAppSelector(state =>
+    selectRelatedAssetIds(state, relatedAssetIdsFilter),
+  )
+
+  const accountIdsByChainId = useAppSelector(selectAccountIdsByChainId)
+
+  const assetsWithFoxOnEthereum = useMemo(() => {
+    return [foxAssetId, ...relatedAssetIds]
+  }, [relatedAssetIds])
+
+  const filteredAssets = useMemo(() => {
+    if (selectedFilters.includes(ALL_FILTER_KEY)) return assetsWithFoxOnEthereum
+
+    return assetsWithFoxOnEthereum.filter(assetId => {
+      const chainId = fromAssetId(assetId).chainId
+      return selectedFilters.includes(chainId)
+    })
+  }, [selectedFilters, assetsWithFoxOnEthereum])
+
+  const filters = useMemo<Filter[]>(() => {
+    return [
+      {
+        label: ALL_FILTER_KEY,
+      },
+      ...assetsWithFoxOnEthereum.map(assetId => ({
+        label: chainIdToChainDisplayName(fromAssetId(assetId).chainId),
+        chainId: fromAssetId(assetId).chainId,
+        assetId,
+      })),
+    ]
+  }, [assetsWithFoxOnEthereum])
+
+  const handleFilterClick = (filter: Filter) => {
+    setSelectedFilters(prev => {
+      if (filter.label === ALL_FILTER_KEY) {
+        return [ALL_FILTER_KEY]
+      }
+
+      if (!filter.chainId) return prev
+
+      const newFilters = prev.filter(f => f !== ALL_FILTER_KEY && f !== filter.label)
+
+      if (!prev.includes(filter.chainId)) {
+        newFilters.push(filter.chainId)
+      } else {
+        newFilters.splice(newFilters.indexOf(filter.chainId), 1)
+      }
+
+      return newFilters.length ? newFilters : [ALL_FILTER_KEY]
+    })
+  }
+
+  const filteredAssetsAccountsWithBalances = useMemo(() => {
```

Comment:
suggestion: to match the rendered component name
```suggestion
  const filteredAssetsAccountEntryRows = useMemo(() => {
```

Reply:
changed in [e05e4b2](https://github.com/shapeshift/web/pull/7882/commits/e05e4b244591c3a87f0da0fca531273c9a08a4c3)